### ISSUE TITLE
^compile does not resolve correct version of sbt plugin dependency (using addSbtPlugin)

### ISSUE THREAD

**Issue by jeffreyolchovy** [sbt/sbt#3393]
## steps
Given a build.sbt for an sbt plugin that needs to include another sbt plugin as a library dependency (e.g. we are creating an AutoPlugin which requires this other sbt plugin):

```scala
sbtPlugin := true

crossSbtVersions := Seq("0.13.16", "1.0.0-RC3")

addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.7.0")
```

## problem
When attempting to cross-build (`^compile`) this plugin for the given versions of sbt, the resolution for the plugin used as a dependency fails:

```
> ^compile
[info] Setting `sbtVersion in pluginCrossBuild` to 0.13.16
[info] Set current project to sbt-issue (in build file:/Users/jeffo/Projects/jvm/sbt-issue/)
[success] Total time: 2 s, completed Jul 31, 2017 10:09:42 AM
[info] Setting `sbtVersion in pluginCrossBuild` to 1.0.0-RC3
[info] Set current project to sbt-issue (in build file:/Users/jeffo/Projects/jvm/sbt-issue/)
[info] Resolving com.eed3si9n#sbt-buildinfo;0.7.0 ...
[warn] 	module not found: com.eed3si9n#sbt-buildinfo;0.7.0
[logging elided] ...
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: com.eed3si9n#sbt-buildinfo;0.7.0: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn] 	Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn] 		com.eed3si9n:sbt-buildinfo:0.7.0 (scalaVersion=2.12, sbtVersion=0.13)
[warn] 
[warn] 	Note: Unresolved dependencies path:
[warn] 		com.eed3si9n:sbt-buildinfo:0.7.0 (scalaVersion=2.12, sbtVersion=0.13) (/Users/jeffo/Projects/jvm/sbt-issue-1/build.sbt#L5-6)
[warn] 		  +- default:sbt-issue-1:0.1-SNAPSHOT (scalaVersion=2.12, sbtVersion=1.0)
[trace] Stack trace suppressed: run last *:update for the full output.
[error] (*:update) sbt.ResolveException: unresolved dependency: com.eed3si9n#sbt-buildinfo;0.7.0: not found
```
## expectation
I would expect `addSbtPlugin` to use the correct version of Scala, conditionally, for each version of sbt that it is attempting to compile against.

## notes
This can be worked around by replacing `addSbtPlugin` with the following:
```scala
libraryDependencies += {
  val currentSbtVersion = (sbtBinaryVersion in pluginCrossBuild).value
  Defaults.sbtPluginExtra("com.eed3si9n" % "sbt-buildinfo" % "0.7.0", currentSbtVersion, scalaBinaryVersion.value)
}
```

sbt version: 0.13.16, cross building for 0.13.16 and 1.0.0-RC3

**Comment by dwijnand** [comment]
Yeah. `addSbtPlugin` should be redefined to use `in pluginCrossBuild`.

Good issues, @jeffreyolchovy, thank you.

**Comment by alonsodomin** [comment]
this exact same problem can be observed in the recently released SBT 1.0.0, preventing cross compiling plugins that depend on other plugins (the exact same workaround suggested here fixes the error).

Could be this ported to SBT 1.0.1?

**Comment by joan38** [comment]
Is there a PR for this or a commit?
Which version are we targeting?

**Comment by eed3si9n** [comment]
Fix was merged to 0.13 branch - https://github.com/sbt/sbt/pull/3397
Still needs forward porting to 1.x branch

**Comment by jroper** [comment]
For those too impatient to wait for a new release:

https://github.com/sbt/sbt-web-build-base/blob/9503fa2cb8cea57cdd859e140941514766a9aebd/src/main/scala/SbtWebBuildBase.scala#L16-L25
